DigitalLearningSolutions.Data/Helpers/FilteringHelper.cs

```diff
@@ -39,19 +40,20 @@ public static string BuildFilterValueString(string group, string propertyName, s
         public static string? GetFilterString(
             string? existingFilterString,
             string? newFilterToAdd,
+            bool clearFilters,
             HttpRequest request,
             string cookieName,
             string? defaultFilterValue = null
         )
         {
-            if (existingFilterString == null && newFilterToAdd == null)
+            if (existingFilterString == null && newFilterToAdd == null && !clearFilters)
             {
                 return request.Cookies.ContainsKey(cookieName)
-                    ? request.Cookies[cookieName]
+                    ? request.Cookies[cookieName] == EmptyFiltersCookieValue ? null : request.Cookies[cookieName]

```

Comment:
Yes? We only want the default filter to be applied when there is no cookie set, or there is an exception. We explicitly don't want the default filter to come back if we have a cookie equivalent to "filters have been cleared", this is one of the bugs that this ticket is to fix.
HEE have also requested that the 30 day expiry be kept. The only reason that we are setting the cookie to EmptyFiltersCookieValue is because you can't insert a null value when setting the cookies.